index.js

```diff
@@ -7,12 +8,229 @@ function ask(questionText) {
+  );
+
+  // Lowest possible guess, casted to number
+  let min = await ask("\nWhat is your minimum number?\n");
+  while (min < 1) {
+    min = await ask("\nWhat is your minimum number?\n");
+  }
+  min = parseInt(min);
+
+  // Highest possible guess, casted to number
+  let max = await ask("\nWhat is your maximum number?\n");
+  while (max <= min) {
+    max = await ask("\nWhat is your maximum number?\n");
+  }
+  max = parseInt(max);
+
+  // Player's number choice
+  let secretNumber = await ask(
+    "\nWhat is your secret number?\nI won't peek, I promise...\n"
+  );
+
+  // Number of guesses
+  let guesses = 0;
+
+  // If the entry is not within the specified range
+  if (secretNumber < min || secretNumber > max) {
+    secretNumber = await ask(
+      "\nOops! That is not within the range you specified, try again...\n"
+    );
+  }
+
+  console.log(
+    "\nYou entered: " + secretNumber + "\nPress any key to continue...\n"
+  );
+
+  // Function for averaging min and max
+  function average(min, max) {
+    let answer = ((max - min) / 2) + min;
+    answer = Math.floor(answer);
+    return answer;
+  }
+
+  // Prompts a guess
+  process.stdin.once("data", async function guessNumber() {
+    guesses++;
+    let yesNo = await ask(`Is ${average(min, max)} your number? (Y/N)\n`);
+    
+    while (yesNo.toLowerCase() === "n") {
+      // Asks if it is higher or lower
+      let hiLow = await ask(
+        `\nIs it Higher (H) or Lower (L) than ${average(min, max)}\n`
+      );
+
+      // If higher: guess becomes minimum answer, next guess is average between present guess and maximum answer
+      if (hiLow.toLowerCase() === "h") {
+        console.log(secretNumber);
+        console.log(max);
+        console.log(min);
+        if (secretNumber == max) {
+          console.log("\nYou're not telling the truth...\n");
+          yesNo = "y";
+          break;
+        }
+        min = secretNumber;
+      }
+      // If lower: guess becomes maximum answer, next guess is average between present guess and minimum answer
+      else if (hiLow.toLowerCase() === "l") {
+        console.log(secretNumber);
+        console.log(max);
+        console.log(min);
+        if (secretNumber == min) {
+          console.log("\nYou're not telling the truth...\n");
+          yesNo = "y";
+          break;
+        }
+        max = secretNumber;
+      }
+      // Was not an H or L
+      else {
+        console.log("That was not an H or an L...");
+      }
+
+      guesses++;
+
+      // Ask if it is the number again
+      yesNo = await ask(`\nIs ${average(min, max)} your number? (Y/N)\n`);
+    }
+
+    // You win wooooo!
+    if (yesNo.toLowerCase() === "y") {
+      console.log(`\nHooray I am so good at this!\nOnly took ${guesses} guesses!`);
+      rematch();
+    }
+    // Game over :(
+    else {
+      console.log(
+        "\nWhat did you just type???\nYou're done dude\nI can't even...\n"
+      );
+    }
+  });
+}
+
```

Comment:
The reverse game still needs work on its input sanitization/recognizing incorrect input, but other than that its working great!